clang/include/clang/Basic/Attr.td

```diff
@@ -2892,6 +2892,19 @@ def CHERILibCall : DeclOrTypeAttr {
   let Subjects = SubjectList<[Function], ErrorDiag>;
 }
 
+def CHERIMMIODevice : DeclOrTypeAttr {
+  let Spellings = [GNU<"cheriot_mmio">];
+  let Documentation = [Undocumented];
+  let Args = [StringArgument<"DeviceName">, StringArgument<"EncodedPermissions">];
```

Comment:
Perhaps a question for @davidchisnall, but I don't completely understand why we need a device name argument. Why can't we just use the GV's name? Do we need to support them being different?